src/data/pnat/basic.lean

```diff
@@ -190,35 +190,36 @@ end
 
 theorem add_sub_of_lt {a b : ℕ+} : a < b → a + (b - a) = b :=
  λ h, eq $ by { rw [add_coe, sub_coe, if_pos h],
-                exact nat.add_sub_of_le (le_of_lt h) }
+                exact nat.add_sub_of_le h.le }
 
 instance : has_well_founded ℕ+ := ⟨(<), measure_wf coe⟩
 
-/-- Strong induction on `pnat`. -/
-lemma strong_induction_on {p : pnat → Prop} : ∀ (n : pnat) (h : ∀ k, (∀ m, m < k → p m) → p k), p n
+/-- Strong induction on `ℕ+`. -/
+def strong_induction_on {p : ℕ+ → Sort*} : ∀ (n : ℕ+) (h : ∀ k, (∀ m, m < k → p m) → p k), p n
 | n := λ IH, IH _ (λ a h, strong_induction_on a IH)
 using_well_founded { dec_tac := `[assumption] }
 
-/-- If `(n : pnat)` is different from `1`, then it is the successor of some `(k : pnat)`. -/
-lemma exists_eq_succ_of_ne_one : ∀ {n : pnat} (h1 : n ≠ 1), ∃ (k : pnat), n = k + 1
+/-- If `n : ℕ+` is different from `1`, then it is the successor of some `k : ℕ+`. -/
+lemma exists_eq_succ_of_ne_one : ∀ {n : ℕ+} (h1 : n ≠ 1), ∃ (k : ℕ+), n = k + 1
 | ⟨1, _⟩ h1 := false.elim $ h1 rfl
 | ⟨n+2, _⟩ _ := ⟨⟨n+1, by simp⟩, rfl⟩
 
-lemma case_strong_induction_on {p : pnat → Prop} (a : pnat) (hz : p 1)
+def case_strong_induction_on {p : ℕ+ → Sort*} (a : ℕ+) (hz : p 1)
   (hi : ∀ n, (∀ m, m ≤ n → p m) → p (n + 1)) : p a :=
 begin
   apply strong_induction_on a,
-  intros k hk,
-  by_cases h1 : k = 1, { rwa h1 },
-  obtain ⟨b, rfl⟩ := exists_eq_succ_of_ne_one h1,
-  simp only [lt_add_one_iff] at hk,
-  exact hi b hk
+  rintro ⟨k, kprop⟩ hk,
+  cases k with k,
+  { exact (lt_irrefl 0 kprop).elim },
+  cases k with k,
+  { exact hz },
+  exact hi ⟨k.succ, nat.succ_pos _⟩ (λ m hm, hk _ (lt_succ_iff.2 hm)),
 end
 
-/-- An induction principle for `pnat`: it takes values in `Sort*`, so it applies also to Types,
+/-- An induction principle for `ℕ+`: it takes values in `Sort*`, so it applies also to Types,
 not only to `Prop`. -/
 @[elab_as_eliminator]
-def rec_on (n : pnat) {p : pnat → Sort*} (p1 : p 1) (hp : ∀ n, p n → p (n + 1)) : p n :=
+def rec_on (n : ℕ+) {p : ℕ+ → Sort*} (p1 : p 1) (hp : ∀ n, p n → p (n + 1)) : p n :=
```

Comment:
are all the induction principles different? I remember some seeming very similar last time I saw this file

Reply:
Yes they are.
`rec_on` is usual induction
`strong_induction_on` is strong induction
`case_strong_induction_on` is strong induction, but with `n = 1` treated separately.